apps/datapuller/src/course.ts

```diff
@@ -1,43 +1,52 @@
-import { TermModel } from "@repo/common";
-import { ClassesAPI } from "@repo/sis-api/classes";
+import { ICourseItem } from "@repo/common";
 import { CoursesAPI } from "@repo/sis-api/courses";
-import { TermsAPI } from "@repo/sis-api/terms";
 
 import setup from "./shared";
+import { Config } from "./shared/config";
+import mapCourseToNewCourse, { CombinedCourse } from "./shared/courseParser";
+import { fetchActiveTerms, fetchPaginatedData } from "./shared/utils";
 
-async function main() {
-  const { log } = setup();
+async function updateCourses(config: Config) {
+  const log = config.log;
+  const coursesAPI = new CoursesAPI();
+
+  log.info("Fetching Active Terms");
+  const activeTerms = await fetchActiveTerms(log, {
```

Comment:
Courses aren't related to terms, so you don't have to fetch terms in this pipeline